src/mono/wasm/runtime/assets.ts

```diff
@@ -76,72 +126,65 @@ export async function mono_download_assets(): Promise<void> {
     }
 }
 
-export async function start_asset_download(asset: AssetEntry): Promise<AssetEntry | undefined> {
+export async function start_asset_download(asset: AssetEntryInternal, downloadData: boolean): Promise<ArrayBuffer | undefined> {
     try {
-        return await start_asset_download_throttle(asset);
+        return await start_asset_download_throttle(asset, downloadData);
     } catch (err: any) {
+        asset.internalPending = undefined;
         if (err && err.status == 404) {
```

Comment:
How does this `err.status` get populated? When I try locally, I see that we get a `TypeError: Failed to fetch` in `start_asset_download_sources`, but then we throw through `mono_assert` - `Error: Assert failed: Response undefined`.

Can we get the http status, and condition on that? If so, what code do we get in the codespaces case? Maybe we can have a narrow set of cases where we do retries?

Reply:
search for `err.status = response.status;`